mithril-aggregator/src/services/prover.rs

```diff
@@ -157,86 +165,301 @@ mod tests {
+                    let block_number =
+                        block_range.start + (j / max_transaction_per_block_number) as u64;
+                    let slot_number = 100 * block_number;
+                    let immutable_file_number = block_number / 5;
+                    let tx_hash = format!(
+                        "tx-br-{}..{}-{}-idx-{}",
+                        block_range.start, block_range.end, j, transaction_index
+                    );
+                    let block_hash = format!("block_hash-{block_number}");
+                    transactions.push(CardanoTransaction::new(
+                        &tx_hash,
+                        block_number,
+                        slot_number,
+                        block_hash,
+                        immutable_file_number,
+                    ));
+                }
+            }
+
+            transactions
+        }
+
+        pub(crate) fn filter_transactions_for_indices(
+            indices: &[usize],
+            transactions: &[CardanoTransaction],
+        ) -> Vec<CardanoTransaction> {
+            transactions
+                .iter()
+                .enumerate()
+                .filter(|(i, _)| indices.contains(i))
+                .map(|(_, t)| t.to_owned())
+                .collect()
+        }
+
+        pub(crate) fn compute_transaction_hashes_from_transactions(
+            transactions: &[CardanoTransaction],
+        ) -> Vec<TransactionHash> {
+            transactions
+                .iter()
+                .map(|t| t.transaction_hash.clone())
+                .collect()
         }
 
-        (hashes, transactions)
+        pub(crate) fn compute_block_ranges_map_from_transactions(
+            transactions: &[CardanoTransaction],
+        ) -> BTreeMap<BlockRange, Vec<CardanoTransaction>> {
+            let mut block_ranges_map = BTreeMap::new();
+            for transaction in transactions {
+                let block_range = BlockRange::from_block_number(transaction.block_number);
+                let block_range_transactions: &mut Vec<_> =
+                    block_ranges_map.entry(block_range).or_insert(vec![]);
+                block_range_transactions.push(transaction.to_owned())
+            }
+
+            block_ranges_map
+        }
+
+        pub(crate) fn filter_transactions_for_block_ranges(
+            block_ranges: &[BlockRange],
+            transactions: &[CardanoTransaction],
+        ) -> Vec<CardanoTransaction> {
+            transactions
+                .iter()
+                .filter(|t| block_ranges.contains(&BlockRange::from_block_number(t.block_number)))
+                .map(|t| t.to_owned())
+                .collect()
+        }
+
+        pub(crate) fn compute_mk_tree_from_transactions(
+            transactions: &[CardanoTransaction],
+        ) -> MKTree {
+            MKTree::new(&compute_transaction_hashes_from_transactions(transactions)).unwrap()
```

Comment:
If I'm not wrong, since `CardanoTransaction` can be converted into a `MKTreeNode`, you can simplify this to (and maybe remove this method as it will bring less value this way):
```suggestion
            MKTree::new(transactions).unwrap()
```

Reply:
Done